src/main/java/org/openlmis/stockmanagement/web/ValidReasonAssignmentController.java

```diff
@@ -153,7 +153,9 @@ private void checkIsValidRequest(ValidReasonAssignment assignment) {
         .findByProgramIdAndFacilityTypeIdAndReasonId(programId, facilityTypeId, reasonId);
 
     if (foundAssignment != null) {
```

Comment:
Since we are only checking the assignemnt exists, we shouldn't be retrieving the whole object from the database - instead we should use the exists check.